packages/searchbase-js/src/index.js

```diff
@@ -0,0 +1,282 @@
+      this.searchDOM.dataset.loading = this.suggestionsLoading;
+    });
+
+    this.searchBaseInstance.onSuggestions = suggestions => {
+      this.suggestions = suggestions.data;
+      this.updateSuggestions();
+    };
+
+    /* Binding Methods */
+    this.initialize = this.initialize.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleOuterClick = this.handleOuterClick.bind(this);
+    this.handleSuggestionClick = this.handleSuggestionClick.bind(this);
+    this.getSuggestionValue = this.getSuggestionValue.bind(this);
+    this.handleSuggestionMouseDown = this.handleSuggestionMouseDown.bind(this);
+    this.handleSuggestionClick = this.handleSuggestionClick.bind(this);
+    this.updateSuggestions = this.updateSuggestions.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.scrollToSelectedSuggestion = this.scrollToSelectedSuggestion.bind(
+      this
+    );
+
+    /* Autocomplete Speifics */
+    this.renderNoSuggestion = renderNoSuggestion || null;
+    if (typeof renderSuggestion === 'function') {
+      this.renderSuggestion = renderSuggestion;
+    }
+    this.initialize();
+  }
+
+  initialize() {
+    this.searchDOM.style.position = 'relative';
+
+    this.input = this.searchDOM.querySelector('input');
+    this.suggestionList = this.searchDOM.querySelector('ul');
+
+    this.expanded = false;
+    this.selectedIndex = -1;
+
+    /* Attributes & Styles */
+
+    this.input.setAttribute('aria-expanded', this.expanded);
+
+    this.suggestionList.style.position = 'absolute';
+    this.suggestionList.style.zIndex = '1';
+    this.suggestionList.style.width = '100%';
+    this.suggestionList.style.boxSizing = 'border-box';
+
+    this.position = getRelativePosition(this.input, this.suggestionList);
+
+    if (!this.suggestionList.id) {
+      this.suggestionList.id = uniqueId('autocomplete-suggestion-list-');
+    }
+    this.input.setAttribute('aria-owns', this.suggestionList.id);
+
+    /* Intiializing Events */
+    document.addEventListener('click', this.handleOuterClick);
+    this.input.addEventListener('input', this.handleInput);
+    this.input.addEventListener('focus', this.handleFocus);
+    this.input.addEventListener('keydown', this.handleKeyDown);
+
+    this.suggestionList.addEventListener('click', this.handleSuggestionClick);
+    this.suggestionList.addEventListener(
+      'mousedown',
+      this.handleSuggestionMouseDown
+    );
+  }
+
+  // eslint-disable-next-line
+  getSuggestionValue(suggestion) {
+    return suggestion.label;
+  }
+
+  /* Rendering */
+  renderSuggestion(suggestion, props) {
+    return `<li ${props}>${this.getSuggestionValue(suggestion)}</li>`;
+  }
+
+  updateSuggestions() {
+    this.suggestionList.innerHTML = '';
+
+    if (this.suggestions.length === 0 && this.renderNoSuggestion) {
```

Comment:
Don't do that in that way `suggestions` property in `Searchbase` is an object of the `Results` class which has the following properties.
It is missing from searchbase docs, but I adding it here for your reference.
```ts
suggestions: Results;
```

It is an object which contains the following details of `suggestions` query response.

#### data: Array<Suggestion>

`data` property contains the parsed suggestions results, a `Suggestion` object has the following format:

```js
{
  label: string,
  value: string,
  source?: any
};
```

#### raw: Object

Response returned by ES query in the raw form.

#### numberOfResults: number

Total number of results found

#### time: number

Total time taken by request (in ms)

#### promotedData: Array<Object>

An array of promoted results obtained from the applied query.

#### rawData: Array<Object>

An array of original hits obtained from the applied query.